src/api/transactions.js

```diff
@@ -1,5 +1,56 @@
 const express = require('express');
+const { Transaction } = require('../db/models');
+const { getDate } = require('../utils');
 
 const router = express.Router();
 
+/**
+ * Get transactions grouped by date, month, or year between two intervals 
+ * Sample query: /api/transactions/intervals?orderBy=day&startDate=2021-04&endDate=2022-09
+ */
+ router.get('/intervals', async (req, res, next) => {
+    try {
```

Comment:
Rather than wrapping the entire route in one large `try/catch` block, it's a better practice to only put code that we expect to get an error from within the block.